components/list/list-item-mixin.js

```diff
@@ -478,7 +479,7 @@ export const ListItemMixin = superclass => class extends composeMixins(
 	}
 
 	updateParentHasAddButon(addButton, addButtonText) {
-		this._addButtonText = addButtonText || this.localize('components.list-item.addItem');
+		this._addButtonText = addButtonText;
```

Comment:
It turned out that `this.localize` running here was returning an empty string, so I moved it to within the rendering